##### 一 panic 与 recover

> 1. 这俩主要用于错误的抛出和捕获
> 2. `recover`必须 直接 放到`defer`函数体中执行，才可以捕获到
>
> 3. panic还可用于调试日志
> 4. PS: goruntine中的panic 不能被捕获

```go
func main(){
  defer func(){
    if err := recover(); err != nil{
      fmt.Println(err)
    }
  }()
  
  panic("我跑出错误啦")
}
// 我跑出错误啦
```

